### ISSUE TITLE
App Crash on Hot Reload

### ISSUE THREAD

**Issue by SankaCoffee** [flutter/flutter#21374]
My App crashes when hot-reloading, both from IDE or Flutter run.
Vebose output of flutter run after reloading:

[+5501 ms] Initializing hot reload...
[   +5 ms] Refreshing active FlutterViews before reloading.
[        ] _flutter.listViews: {isolateId: isolates/224998240}
[ +276 ms] Syncing files to device Android SDK built for x86...
[        ] DevFS: Starting sync from LocalDirectory: 'C:\Projecten\Van Doren - Introductie App'
[        ] Scanning project files
[   +1 ms] Scanning package files
[  +30 ms] Scanning asset files
[        ] Scanning for deleted files
[  +26 ms] Compiling dart to kernel with 0 updated files
[  +10 ms] Updating files
[   +9 ms] DevFS: Sync finished
[        ] Syncing files to device Android SDK built for x86... (completed)
[        ] Synced 0.0MB.
[   +2 ms] _flutter.setAssetBundlePath: {isolateId: isolates/224998240, viewId: _flutterView/0xec23760c, assetDirectory: /data/user/0/com.example.introductieapp/cache/Van Doren - Introductie AppGZIMCN/Van Doren - Introductie App/build/flutter_assets}
[   +8 ms] _reloadSources: {pause: false, rootLibUri: file:///data/user/0/com.example.introductieapp/cache/Van%20Doren%20-%20Introductie%20AppGZIMCN/Van%20Doren%20-%20Introductie%20App/lib/main.dart.incremental.dill, packagesUri: file:///data/user/0/com.example.introductieapp/cache/Van%20Doren%20-%20Introductie%20AppGZIMCN/Van%20Doren%20-%20Introductie%20App/.packages, isolateId: isolates/224998240}
[   +9 ms] D/        (26871): HostConnection::get() New Host Connection established 0xe43e3080, tid 26889
[  +82 ms] Service protocol connection closed.
[   +2 ms] Lost connection to device.
[   +3 ms] DevFS: Deleting filesystem on the device (file:///data/user/0/com.example.introductieapp/cache/Van%20Doren%20-%20Introductie%20AppGZIMCN/Van%20Doren%20-%20Introductie%20App/)
[   +1 ms] _deleteDevFS: {fsName: Van Doren - Introductie App}
[ +255 ms] TimeoutException after 0:00:00.250000: Future not completed

**Comment by mhazy** [comment]
Are you able to provide a sample of code to reproduce this?

**Comment by SankaCoffee** [comment]
It doesn't happen when i create a new project. 
I'll try to narrow it down more, though it happens on more than 1 project.

**Comment by SankaCoffee** [comment]
I reduced my App code to this, and the error still persists:

```
import 'package:flutter/material.dart';

void main() {
  runApp(VDApp());
}

class VDApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      // navigatorKey: navigatorKey,
      onGenerateTitle: (BuildContext context) => "Test",
      onGenerateRoute: (RouteSettings settings) {
        return MaterialPageRoute(builder: (context) {
          List<Widget> children = <Widget>[];
          // if (vanDorenStore.state.currentPage.background != null)
          children.add(Container(color: Colors.green));

          // if (vanDorenStore.state.currentPage.content != null)
          children.add(Container(child:Text("Test test ")));

          return Stack(
            children: children
          );
        });
        // return VanDorenPageRoute(pageData: vanDorenStore.state.currentPage);
      }
    );
  }
}
```
However, copy/paste this code into a newly created app doesn't crash on reload.
I'll try re-creating the app and copying the .dart source files into it.

**Comment by zoechi** [comment]
Have you tried `flutter clean`?

**Comment by SankaCoffee** [comment]
Yeah, tried that. 
I think it has something to do with the name of the directory, which containes a dash.
The project is in source-control and cloned into a directory which does not match the app-name inside pubspec.yaml. Renaming the directory to not contain spaces and/or dashes seems to have fixed it.

Should the directory-name reflect the pubspec.yaml name? Or just a name without spaces / glyphs?

**Comment by jason-simmons** [comment]
This should be fixed in the next engine roll (see https://fuchsia.googlesource.com/tonic/+/3c566bc96b527f7032969b23fc1ddadc6fd165af)

**Comment by zoechi** [comment]
@SankaCoffee are you still seeing this issue?

**Comment by SankaCoffee** [comment]
@zoechi No, it seems fixed. I also renamed my working directory with several special chars, but it didn't crash anymore. Thanks!
